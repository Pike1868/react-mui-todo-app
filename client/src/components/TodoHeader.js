import React, { useState } from "react";
import { AppBar, Box, Button, Typography } from "@mui/material";
import Modal from "./Modal";
import { useCookies } from "react-cookie";

const TodoHeader = ({ listName, getData }) => {
  const [showModal, setShowModal] = useState(false);
  const [cookies, setCookie, removeCookie] = useCookies(null);

  const signOut = () => {
    console.log("Sign out button clicked");
    removeCookie("Email");
    removeCookie("AuthToken");

    // window.location.reload();
  };

  return (
    <AppBar
      position="static"
      color=""
      sx={{
        p: 1,
        display: "flex",
        flexDirection: "row",
        justifyContent: "space-between",
        marginTop: "1rem",
        backgroundColor: "#F1F1F1",
        borderRadius: "0",
      }}
    >
      <Typography variant="h6" color="warning" sx={{ p: 2 }}>
        {listName}
      </Typography>
      <Box sx={{ display: "flex", justifyContent: "space-evenly" }}>
        <Button
          variant="contained"
          color="success"
          sx={{ m: 2, borderRadius: 5, fontSize: "0.8rem" }}
          onClick={() => setShowModal(true)}
        >
          <Typography variant="body">ADD NEW</Typography>
        </Button>
        <Button
          variant="contained"
          color="error"
          onClick={signOut}
          sx={{ m: 2, borderRadius: 5, fontSize: "0.8rem" }}
        >
          Sign Out
        </Button>
      </Box>
      {showModal && (
        <Modal mode={"create"} setShowModal={setShowModal} getData={getData} />
      )}
    </AppBar>
  );
};

export default TodoHeader;
