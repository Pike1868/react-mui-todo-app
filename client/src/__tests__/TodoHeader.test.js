// TodoHeader.test.js
import React from "react";
import { render } from "@testing-library/react";
import TodoHeader from "../components/TodoHeader";

// Smoke Test
it("renders without crashing", () => {
  render(<TodoHeader listName="My Todo List" getData={() => {}} />);
});

// Snapshot Test
it("matches snapshot", () => {
  const { asFragment } = render(
    <TodoHeader listName="My Todo List" getData={() => {}} />
  );
  expect(asFragment()).toMatchSnapshot();
});
