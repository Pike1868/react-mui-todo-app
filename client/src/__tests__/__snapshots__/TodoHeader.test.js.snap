// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-color MuiAppBar-positionStatic css-4t15h5-MuiPaper-root-MuiAppBar-root"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-1vhzle8-MuiTypography-root"
    >
      My Todo List
    </h6>
    <div
      class="MuiBox-root css-102lqhp"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-1a1w8ve-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
        >
          ADD NEW
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium css-nxsgnf-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Sign Out
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</DocumentFragment>
`;
